PR: https://github.com/odoo/odoo/pull/236895

From: 3852885f37aa691149f0be7d05d0ef9add91c163
From: Stéphane Diez
Date: 2026-01-23 13:16:51

Breaking data model changes scores: del:7 + add:0, change matches:
-    product_uom_id = fields.Many2one(
-    product_uom_id = fields.Many2one(
-    product_uom_id = fields.Many2one('uom.uom', 'Unit', required=True,
-    product_uom_id = fields.Many2one(
-    product_uom_id = fields.Many2one(
-    product_uom_id = fields.Many2one('uom.uom', string='Unit',
-    product_uom_id = fields.Many2one(related='production_id.product_uom_id')

Total Changes: 408

[IMP] uom: rename logistic fields

Currently, the uom field can have 3 different names in logistic modules:
* uom_id
* product_uom_id
* product_uom

This is annoying when we need to use a uom field and we have to
check what is its name. So we decided to only keep uom_id as the
only name for basic uom fields.

task 5364113

Part-of: odoo/odoo#236895
Related: odoo/enterprise#100172
Related: odoo/upgrade#8959
Signed-off-by: Arnold Moyaux (arm) <arm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mrp/models/mrp_bom.py
+++ b/addons/mrp/models/mrp_bom.py
@@ -18,7 +18,7 @@ class MrpBom(models.Model):
     _order = "sequence, id"
     _check_company_auto = True
 
-    def _get_default_product_uom_id(self):
+    def _default_uom_id(self):
         return self.env['uom.uom'].search([], limit=1, order='id').id
 
     code = fields.Char('Reference')
@@ -42,9 +42,9 @@ class MrpBom(models.Model):
         'Quantity', default=1.0,
         digits='Product Unit', required=True,
         help="This should be the smallest quantity that this product can be produced in. If the BOM contains operations, make sure the work center capacity is accurate.")
-    product_uom_id = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', 'Unit',
-        default=_get_default_product_uom_id, required=True,
+        default=_default_uom_id, required=True,
         help="Unit of Measure (Unit of Measure) is the unit of measurement for the inventory control")
     sequence = fields.Integer('Sequence')
     operation_ids = fields.One2many('mrp.routing.workcenter', 'bom_id', 'Operations', copy=True)
@@ -231,10 +231,10 @@ class MrpBom(models.Model):
                     'message': _("Changing the product or variant will permanently reset all previously encoded variant-related data."),
                 }
             }
-            default_uom_id = self.env.context.get('default_product_uom_id')
+            default_uom_id = self.env.context.get('default_uom_id')
             # Avoids updating the BoM's UoM in case a specific UoM was passed through as a default value.
-            if self.product_uom_id.id != default_uom_id:
-                self.product_uom_id = self.product_tmpl_id.uom_id.id
+            if self.uom_id.id != default_uom_id:
+                self.uom_id = self.product_tmpl_id.uom_id.id
             if self.product_id.product_tmpl_id != self.product_tmpl_id:
                 self.product_id = False
             self.bom_line_ids.bom_product_template_attribute_value_ids = False
@@ -317,8 +317,8 @@ class MrpBom(models.Model):
     def _compute_display_name(self):
         for bom in self:
             display_name = f"{bom.code + ': ' if bom.code else ''}{bom.product_tmpl_id.display_name}"
-            if self.env.context.get('display_bom_uom_qty') and (bom.product_qty > 1 or bom.product_uom_id != bom.product_tmpl_id.uom_id):
-                display_name += f" ({bom.product_qty} {bom.product_uom_id.name})"
+            if self.env.context.get('display_bom_uom_qty') and (bom.product_qty > 1 or bom.uom_id != bom.product_tmpl_id.uom_id):
+                display_name += f" ({bom.product_qty} {bom.uom_id.name})"
             bom.display_name = _('%(display_name)s', display_name=display_name)
 
     @api.depends('operation_ids')
@@ -358,7 +358,7 @@ class MrpBom(models.Model):
 
     @api.constrains('enable_batch_size', 'batch_size')
     def _check_valid_batch_size(self):
-        if any(bom.enable_batch_size and bom.product_uom_id.compare(bom.batch_size, 0.0) <= 0 for bom in self):
+        if any(bom.enable_batch_size and bom.uom_id.compare(bom.batch_size, 0.0) <= 0 for bom in self):
             raise ValidationError(self.env._("The batch size must be positive!"))
 
     @api.ondelete(at_uninstall=False)
@@ -452,8 +452,8 @@ class MrpBom(models.Model):
                 product_ids.clear()
             bom = product_boms.get(current_line.product_id)
             if bom:
-                converted_line_quantity = current_line.product_uom_id._compute_quantity(
-                    line_quantity / bom.product_qty, bom.product_uom_id, round=False
+                converted_line_quantity = current_line.uom_id._compute_quantity(
+                    line_quantity / bom.product_qty, bom.uom_id, round=False
                 )
                 bom_lines = [(line, current_line.product_id, converted_line_quantity, current_line) for line in bom.bom_line_ids] + bom_lines
                 for bom_line in bom.bom_line_ids:
@@ -463,7 +463,7 @@ class MrpBom(models.Model):
             else:
                 # We round up here because the user expects that if he has to consume a little more, the whole UOM unit
                 # should be consumed.
-                line_quantity = current_line.product_uom_id.round(line_quantity, rounding_method='UP')
+                line_quantity = current_line.uom_id.round(line_quantity, rounding_method='UP')
                 lines_done.append((current_line, current_line._prepare_line_done_values(line_quantity, current_product, quantity, parent_line, boms_done)))
 
         lines_done = self._round_last_line_done(lines_done)
@@ -642,7 +642,7 @@ class MrpBom(models.Model):
             ])
             orderpoint.route_id = self.env['stock.rule'].search(domain, limit=1).route_id.id
         orderpoint.bom_id = self
-        bom_qty = self.product_uom_id._compute_quantity(self.product_qty, orderpoint.product_id.uom_id)
+        bom_qty = self.uom_id._compute_quantity(self.product_qty, orderpoint.product_id.uom_id)
         if orderpoint.qty_to_order < bom_qty:
             orderpoint.qty_to_order = bom_qty
         return orderpoint.action_stock_replenishment_info()
@@ -672,7 +672,7 @@ class MrpBomLine(models.Model):
     _description = 'Bill of Material Line'
     _check_company_auto = True
 
-    def _get_default_product_uom_id(self):
+    def _default_uom_id(self):
         return self.env['uom.uom'].search([], limit=1, order='id').id
 
     product_id = fields.Many2one('product.product', 'Component', required=True, check_company=True, index=True)
@@ -682,9 +682,9 @@ class MrpBomLine(models.Model):
     product_qty = fields.Float(
         'Quantity', default=1.0,
         digits='Product Unit', required=True)
-    product_uom_id = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', 'Unit',
-        default=_get_default_product_uom_id, required=True)
+        default=_default_uom_id, required=True)
     sequence = fields.Integer(
         'Sequence', default=1,
         help="Gives the sequence order when displaying.")
@@ -744,13 +744,13 @@ class MrpBomLine(models.Model):
     @api.onchange('product_id')
     def onchange_product_id(self):
         if self.product_id:
-            self.product_uom_id = self.product_id.uom_id.id
+            self.uom_id = self.product_id.uom_id.id
 
     @api.model_create_multi
     def create(self, vals_list):
         for values in vals_list:
-            if 'product_id' in values and 'product_uom_id' not in values:
-                values['product_uom_id'] = self.env['product.product'].browse(values['product_id']).uom_id.id
+            if 'product_id' in values and 'uom_id' not in values:
+                values['uom_id'] = self.env['product.product'].browse(values['product_id']).uom_id.id
         return super(MrpBomLine, self).create(vals_list)
 
     def _skip_bom_line(self, product, never_attribute_values=False):
@@ -817,14 +817,14 @@ class MrpBomLine(models.Model):
                 **self[0].bom_id._get_product_price_and_data(self[0].product_id),
                 'quantity': sum(
                     self.mapped(
-                        lambda line: line.product_uom_id._compute_quantity(
+                        lambda line: line.uom_id._compute_quantity(
                             qty=line.product_qty,
-                            to_unit=line.product_uom_id,
+                            to_unit=line.uom_id,
                         )
                     )
                 ),
                 'readOnly': len(self) > 1,
-                'uomDisplayName': len(self) == 1 and self.product_uom_id.display_name or self.product_id.uom_id.display_name,
+                'uomDisplayName': len(self) == 1 and self.uom_id.display_name or self.product_id.uom_id.display_name,
             }
         return {
             'quantity': 0,
@@ -849,8 +849,8 @@ class MrpBomByproduct(models.Model):
     product_qty = fields.Float(
         'Quantity',
         default=1.0, digits='Product Unit', required=True)
-    product_uom_id = fields.Many2one('uom.uom', 'Unit', required=True,
-                                     compute="_compute_product_uom_id", store=True, readonly=False, precompute=True)
+    uom_id = fields.Many2one('uom.uom', 'Unit', required=True,
+                                     compute="_compute_uom_id", store=True, readonly=False, precompute=True)
     bom_id = fields.Many2one('mrp.bom', 'BoM', ondelete='cascade', index=True)
     allowed_operation_ids = fields.One2many('mrp.routing.workcenter', related='bom_id.operation_ids')
     operation_id = fields.Many2one(
@@ -868,10 +868,10 @@ class MrpBomByproduct(models.Model):
              "The total of all by-products' cost share must be less than or equal to 100.")
 
     @api.depends('product_id')
-    def _compute_product_uom_id(self):
+    def _compute_uom_id(self):
         """ Changes UoM if product_id changes. """
         for record in self:
-            record.product_uom_id = record.product_id.uom_id.id
+            record.uom_id = record.product_id.uom_id.id
 
     def _skip_byproduct_line(self, product, never_attribute_values=False):
         """ Control if a byproduct line should be produced, can be inherited to add
@@ -898,14 +898,14 @@ class MrpBomByproduct(models.Model):
                 **self[0].bom_id._get_product_price_and_data(self[0].product_id),
                 'quantity': sum(
                     self.mapped(
-                        lambda line: line.product_uom_id._compute_quantity(
+                        lambda line: line.uom_id._compute_quantity(
                             qty=line.product_qty,
-                            to_unit=line.product_uom_id,
+                            to_unit=line.uom_id,
                         )
                     )
                 ),
                 'readOnly': len(self) > 1,
-                'uomDisplayName': len(self) == 1 and self.product_uom_id.display_name or self.product_id.uom_id.display_name,
+                'uomDisplayName': len(self) == 1 and self.uom_id.display_name or self.product_id.uom_id.display_name,
             }
         return {
             'quantity': 0,

--- a/addons/mrp/models/mrp_production.py
+++ b/addons/mrp/models/mrp_production.py
@@ -114,7 +114,7 @@ class MrpProduction(models.Model):
         readonly=False, required=True, tracking=True, precompute=True,
         compute='_compute_product_qty', store=True, copy=True)
     allowed_uom_ids = fields.Many2many('uom.uom', compute='_compute_allowed_uom_ids')
-    product_uom_id = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', 'Unit', domain="[('id', 'in', allowed_uom_ids)]",
         readonly=False, required=True, compute='_compute_uom_id', store=True, copy=True, precompute=True)
     lot_producing_ids = fields.Many2many(
@@ -359,11 +359,11 @@ class MrpProduction(models.Model):
             if production.state != 'draft':
                 continue
             if production.bom_id and production._origin.bom_id != production.bom_id:
-                production.product_uom_id = production.bom_id.product_uom_id
+                production.uom_id = production.bom_id.uom_id
             elif production.product_id:
-                production.product_uom_id = production.product_id.uom_id
+                production.uom_id = production.product_id.uom_id
             else:
-                production.product_uom_id = False
+                production.uom_id = False
 
     @api.depends('picking_type_id')
     def _compute_locations(self):
@@ -426,10 +426,10 @@ class MrpProduction(models.Model):
             ):
                 production.product_id = bom.product_id or bom.product_tmpl_id.product_variant_id
 
-    @api.depends('product_id', 'product_id.uom_id', 'product_id.uom_ids', 'product_id.bom_ids', 'product_id.bom_ids.product_uom_id')
+    @api.depends('product_id', 'product_id.uom_id', 'product_id.uom_ids', 'product_id.bom_ids', 'product_id.bom_ids.uom_id')
     def _compute_allowed_uom_ids(self):
         for production in self:
-            production.allowed_uom_ids = production.product_id.uom_id | production.product_id.uom_ids | production.product_id.bom_ids.product_uom_id
+            production.allowed_uom_ids = production.product_id.uom_id | production.product_id.uom_ids | production.product_id.bom_ids.uom_id
 
     @api.depends('product_id', 'never_product_template_attribute_value_ids')
     def _compute_bom_id(self):
@@ -466,7 +466,7 @@ class MrpProduction(models.Model):
             production.production_capacity = production.product_qty
             moves = production.move_raw_ids.filtered(lambda move: move.unit_factor and move.product_id.type != 'consu')
             if moves:
-                production_capacity = min(moves.mapped(lambda move: move.product_id.uom_id._compute_quantity(move.product_id.qty_available, move.product_uom) / move.unit_factor))
+                production_capacity = min(moves.mapped(lambda move: move.product_id.uom_id._compute_quantity(move.product_id.qty_available, move.uom_id) / move.unit_factor))
                 production.production_capacity = min(production.product_qty, production.product_id.uom_id.round(production_capacity))
 
     @api.depends('move_finished_ids.date_deadline')
@@ -529,11 +529,11 @@ class MrpProduction(models.Model):
             order.picking_ids = move_per_production.get(order.production_group_id, False)
             order.delivery_count = len(order.picking_ids)
 
-    @api.depends('product_uom_id', 'product_qty', 'product_id.uom_id')
+    @api.depends('uom_id', 'product_qty', 'product_id.uom_id')
     def _compute_product_uom_qty(self):
         for production in self:
-            if production.product_id.uom_id != production.product_uom_id:
-                production.product_uom_qty = production.product_uom_id._compute_quantity(production.product_qty, production.product_id.uom_id)
+            if production.product_id.uom_id != production.uom_id:
+                production.product_uom_qty = production.uom_id._compute_quantity(production.product_qty, production.product_id.uom_id)
             else:
                 production.product_uom_qty = production.product_qty
 
@@ -579,7 +579,7 @@ class MrpProduction(models.Model):
         produce and all work orders has been finished.
         """
         for production in self:
-            if not production.state or not production.product_uom_id or not (production.id or production._origin.id):
+            if not production.state or not production.uom_id or not (production.id or production._origin.id):
                 production.state = 'draft'
             elif production.state == 'cancel' or (production.move_finished_ids and all(move.state == 'cancel' for move in production.move_finished_ids)):
                 production.state = 'cancel'
@@ -591,16 +591,16 @@ class MrpProduction(models.Model):
                 production.state = 'done'
             elif production.workorder_ids and all(wo_state in ('done', 'cancel') for wo_state in production.workorder_ids.mapped('state')):
                 production.state = 'to_close'
-            elif not production.workorder_ids and production.product_uom_id.compare(production.qty_producing, production.product_qty) >= 0:
+            elif not production.workorder_ids and production.uom_id.compare(production.qty_producing, production.product_qty) >= 0:
                 production.state = 'to_close'
             elif (
                 any(wo_state in ('progress', 'done') for wo_state in production.workorder_ids.mapped('state'))
-                or production.product_uom_id and not production.product_uom_id.is_zero(production.qty_producing)
+                or production.uom_id and not production.uom_id.is_zero(production.qty_producing)
                 or any(production.move_raw_ids.mapped('picked'))
             ):
                 production.state = 'progress'
 
-    @api.depends('bom_id', 'product_id', 'product_qty', 'product_uom_id', 'never_product_template_attribute_value_ids')
+    @api.depends('bom_id', 'product_id', 'product_qty', 'uom_id', 'never_product_template_attribute_value_ids')
     def _compute_workorder_ids(self):
         for production in self:
             if production.state != 'draft':
@@ -621,7 +621,7 @@ class MrpProduction(models.Model):
             if production.bom_id and production.product_id and production.product_qty > 0:
                 # keep manual entries
                 workorders_values = []
-                product_qty = production.product_uom_id._compute_quantity(production.product_qty, production.bom_id.product_uom_id)
+                product_qty = production.uom_id._compute_quantity(production.product_qty, production.bom_id.uom_id)
                 exploded_boms, _dummy = production.bom_id.explode(production.product_id, product_qty / production.bom_id.product_qty, picking_type=production.bom_id.picking_type_id,
                     never_attribute_values=production.never_product_template_attribute_value_ids)
 
@@ -640,7 +640,7 @@ class MrpProduction(models.Model):
                             'name': operation.name,
                             'production_id': production.id,
                             'workcenter_id': operation.workcenter_id.id,
-                            'product_uom_id': production.product_uom_id.id,
+                            'uom_id': production.uom_id.id,
                             'operation_id': operation.id,
                             'state': 'ready',
                         }]
@@ -668,7 +668,7 @@ class MrpProduction(models.Model):
                     m.product_id
                     and not (
                         m.picked
-                        or m.product_uom.is_zero(
+                        or m.uom_id.is_zero(
                             m.product_uom_qty,
                         )
                     )
@@ -794,7 +794,7 @@ class MrpProduction(models.Model):
                 date_finished = date_finished + relativedelta(minutes=workorder_expected_duration or 60)
             production.date_finished = date_finished
 
-    @api.depends('company_id', 'bom_id', 'product_id', 'product_qty', 'product_uom_id', 'location_src_id', 'never_product_template_attribute_value_ids')
+    @api.depends('company_id', 'bom_id', 'product_id', 'product_qty', 'uom_id', 'location_src_id', 'never_product_template_attribute_value_ids')
     def _compute_move_raw_ids(self):
         for production in self:
             if production.state != 'draft' or self.env.context.get('skip_compute_move_raw_ids'):
@@ -820,7 +820,7 @@ class MrpProduction(models.Model):
             else:
                 production.move_raw_ids = [Command.delete(move.id) for move in production.move_raw_ids.filtered(lambda m: m.bom_line_id)]
 
-    @api.depends('product_id', 'bom_id', 'product_qty', 'product_uom_id', 'location_dest_id', 'date_finished', 'move_dest_ids', 'never_product_template_attribute_value_ids')
+    @api.depends('product_id', 'bom_id', 'product_qty', 'uom_id', 'location_dest_id', 'date_finished', 'move_dest_ids', 'never_product_template_attribute_value_ids')
     def _compute_move_finished_ids(self):
         production_with_move_finished_ids_to_unlink_ids = OrderedSet()
         ignored_mo_ids = self.env.context.get('ignore_mo_ids', [])
@@ -1133,7 +1133,7 @@ class MrpProduction(models.Model):
             'default_product_id': self.product_id.id,
             'default_product_qty': self.product_qty,
             'default_product_tmpl_id': self.product_id.product_tmpl_id.id,
-            'default_product_uom_id': self.product_uom_id.id,
+            'default_uom_id': self.uom_id.id,
             'parent_production_id': self.id,  # Used to assign the new BoM to the current MO.
         }
         return action
@@ -1195,11 +1195,11 @@ class MrpProduction(models.Model):
 
         def get_uom_and_quantity(move):
             # Use the BoM line/by-product's UoM if the move is linked to one of them.
-            target_uom = (move.bom_line_id or move.byproduct_id).product_uom_id or move.product_uom
+            target_uom = (move.bom_line_id or move.byproduct_id).uom_id or move.uom_id
             # In order to be able to multiply the move quantity by the ratio, we
             # have to be sure they both express in the same UoM.
             qty = move.quantity or move.product_uom_qty
-            qty = move.product_uom._compute_quantity(qty * ratio, target_uom)
+            qty = move.uom_id._compute_quantity(qty * ratio, target_uom)
             return (target_uom, qty)
 
         # BoM lines values.
@@ -1209,7 +1209,7 @@ class MrpProduction(models.Model):
             bom_line_vals = {
                 'product_id': move_raw.product_id.id,
                 'product_qty': qty,
-                'product_uom_id': uom.id,
+                'uom_id': uom.id,
             }
             bom_lines_values.append(Command.create(bom_line_vals))
         # By-Product lines values.
@@ -1220,7 +1220,7 @@ class MrpProduction(models.Model):
                 'cost_share': move_byproduct.cost_share,
                 'product_id': move_byproduct.product_id.id,
                 'product_qty': qty,
-                'product_uom_id': uom.id,
+                'uom_id': uom.id,
             }
             byproduct_values.append(Command.create(bom_byproduct_vals))
         # Operations values.
@@ -1242,7 +1242,7 @@ class MrpProduction(models.Model):
         return {
             'product_id': product_id,
             'product_uom_qty': product_uom_qty,
-            'product_uom': product_uom,
+            'uom_id': product_uom,
             'operation_id': operation_id,
             'byproduct_id': byproduct_id,
             'date': self.date_finished,
@@ -1266,16 +1266,16 @@ class MrpProduction(models.Model):
         for production in self:
             if production.product_id in production.bom_id.byproduct_ids.mapped('product_id'):
                 raise UserError(_("You cannot have %s  as the finished product and in the Byproducts", self.product_id.name))
-            finished_move_values = production._get_move_finished_values(production.product_id.id, production.product_qty, production.product_uom_id.id)
+            finished_move_values = production._get_move_finished_values(production.product_id.id, production.product_qty, production.uom_id.id)
             finished_move_values['location_final_id'] = self.location_final_id.id
             moves.append(finished_move_values)
             for byproduct in production.bom_id.byproduct_ids:
                 if byproduct._skip_byproduct_line(production.product_id, production.never_product_template_attribute_value_ids):
                     continue
-                product_uom_factor = production.product_uom_id._compute_quantity(production.product_qty, production.bom_id.product_uom_id)
+                product_uom_factor = production.uom_id._compute_quantity(production.product_qty, production.bom_id.uom_id)
                 qty = byproduct.product_qty * (product_uom_factor / production.bom_id.product_qty)
                 moves.append(production._get_move_finished_values(
-                    byproduct.product_id.id, qty, byproduct.product_uom_id.id,
+                    byproduct.product_id.id, qty, byproduct.uom_id.id,
                     byproduct.operation_id.id, byproduct.id, byproduct.cost_share))
         return moves
 
@@ -1304,7 +1304,7 @@ class MrpProduction(models.Model):
         for production in self:
             if not production.bom_id:
                 continue
-            factor = production.product_uom_id._compute_quantity(production.product_qty, production.bom_id.product_uom_id, round=False) / production.bom_id.product_qty
+            factor = production.uom_id._compute_quantity(production.product_qty, production.bom_id.uom_id, round=False) / production.bom_id.product_qty
             _boms, lines = production.bom_id.explode(production.product_id, factor, picking_type=production.bom_id.picking_type_id, never_attribute_values=production.never_product_template_attribute_value_ids)
             for bom_line, line_data in lines:
                 if bom_line.child_bom_id and bom_line.child_bom_id.type == 'phantom' or\
@@ -1314,7 +1314,7 @@ class MrpProduction(models.Model):
                 moves.append(production._get_move_raw_values(
                     bom_line.product_id,
                     line_data['qty'],
-                    bom_line.product_uom_id,
+                    bom_line.uom_id,
                     operation,
                     bom_line
                 ))
@@ -1335,7 +1335,7 @@ class MrpProduction(models.Model):
             'picking_type_id': self.picking_type_id.id,
             'product_id': product.id,
             'product_uom_qty': product_uom_qty,
-            'product_uom': product_uom.id,
+            'uom_id': product_uom.id,
             'location_id': source_location.id,
             'location_dest_id': self.product_id.with_company(self.company_id).property_stock_production.id,
             'raw_material_production_id': self.id,
@@ -1364,11 +1364,11 @@ class MrpProduction(models.Model):
 
     def _set_qty_producing(self, pick_manual_consumption_moves=True):
         if self.product_id.tracking == 'serial':
-            qty_producing_uom = self.product_uom_id._compute_quantity(self.qty_producing, self.product_id.uom_id, rounding_method='HALF-UP')
-            qty_production_uom = self.product_uom_id._compute_quantity(self.product_qty, self.product_id.uom_id, rounding_method='HALF-UP')
+            qty_producing_uom = self.uom_id._compute_quantity(self.qty_producing, self.product_id.uom_id, rounding_method='HALF-UP')
+            qty_production_uom = self.uom_id._compute_quantity(self.product_qty, self.product_id.uom_id, rounding_method='HALF-UP')
             # allow changing a non-zero value to a 0 to not block mass produce feature
             if qty_producing_uom != qty_production_uom and not (qty_producing_uom == 0 and self._origin.qty_producing != self.qty_producing):
-                self.qty_producing = self.product_id.uom_id._compute_quantity(len(self.lot_producing_ids), self.product_uom_id, rounding_method='HALF-UP')
+                self.qty_producing = self.product_id.uom_id._compute_quantity(len(self.lot_producing_ids), self.uom_id, rounding_method='HALF-UP')
 
         # waiting for a preproduction move before assignement
         is_waiting = self.warehouse_id.manufacture_steps != 'mrp_one_step' and self.picking_ids.filtered(lambda p: p.picking_type_id == self.warehouse_id.pbm_type_id and p.state not in ('done', 'cancel'))
@@ -1386,7 +1386,7 @@ class MrpProduction(models.Model):
             if move.sudo()._should_bypass_set_qty_producing():
                 continue
 
-            new_qty = move.product_uom.round((self.qty_producing - self.qty_produced) * move.unit_factor)
+            new_qty = move.uom_id.round((self.qty_producing - self.qty_produced) * move.unit_factor)
             move._set_quantity_done(new_qty)
             if (not move.manual_consumption or pick_manual_consumption_moves) \
                     and move.quantity \
@@ -1403,7 +1403,7 @@ class MrpProduction(models.Model):
         update_info = []
         for move in self.move_raw_ids.filtered(lambda m: m.state not in ('done', 'cancel')):
             old_qty = move.product_uom_qty
-            new_qty = move.product_uom.round(old_qty * factor, rounding_method='UP')
+            new_qty = move.uom_id.round(old_qty * factor, rounding_method='UP')
             if new_qty > 0:
                 # procurement and assigning is now run in write
                 move.write({'product_uom_qty': new_qty})
@@ -1584,15 +1584,15 @@ class MrpProduction(models.Model):
             if production.bom_id:
                 production_vals.update({'consumption': production.bom_id.consumption})
             # In case of Serial number tracking, force the UoM to the UoM of product
-            if production.product_tracking == 'serial' and production.product_uom_id != production.product_id.uom_id:
+            if production.product_tracking == 'serial' and production.uom_id != production.product_id.uom_id:
                 production_vals.update({
-                    'product_qty': production.product_uom_id._compute_quantity(production.product_qty, production.product_id.uom_id),
-                    'product_uom_id': production.product_id.uom_id
+                    'product_qty': production.uom_id._compute_quantity(production.product_qty, production.product_id.uom_id),
+                    'uom_id': production.product_id.uom_id
                 })
                 for move_finish in production.move_finished_ids.filtered(lambda m: m.product_id == production.product_id):
                     move_finish.write({
-                        'product_uom_qty': move_finish.product_uom._compute_quantity(move_finish.product_uom_qty, move_finish.product_id.uom_id),
-                        'product_uom': move_finish.product_id.uom_id
+                        'product_uom_qty': move_finish.uom_id._compute_quantity(move_finish.product_uom_qty, move_finish.product_id.uom_id),
+                        'uom_id': move_finish.product_id.uom_id
                     })
             if production_vals:
                 production.write(production_vals)
@@ -1745,13 +1745,13 @@ class MrpProduction(models.Model):
             expected_qty_by_product = defaultdict(float)
             for move_values in expected_move_values:
                 move_product = self.env['product.product'].browse(move_values['product_id'])
-                move_uom = self.env['uom.uom'].browse(move_values['product_uom'])
+                move_uom = self.env['uom.uom'].browse(move_values['uom_id'])
                 move_product_qty = move_uom._compute_quantity(move_values['product_uom_qty'], move_product.uom_id)
                 expected_qty_by_product[move_product] += move_product_qty * order.qty_producing / order.product_qty
 
             done_qty_by_product = defaultdict(float)
             for move in order.move_raw_ids:
-                quantity = move.product_uom._compute_quantity(move._get_picked_quantity(), move.product_id.uom_id)
+                quantity = move.uom_id._compute_quantity(move._get_picked_quantity(), move.product_id.uom_id)
                 # extra lines with non-zero qty picked
                 if move.product_id not in expected_qty_by_product and move.picked and not move.product_id.uom_id.is_zero(quantity):
                     issues.append((order, move.product_id, quantity, 0.0))
@@ -1774,7 +1774,7 @@ class MrpProduction(models.Model):
                 'mrp_production_id': order.id,
                 'product_id': product_id.id,
                 'consumption': order.consumption,
-                'product_uom_id': product_id.uom_id.id,
+                'uom_id': product_id.uom_id.id,
                 'product_consumed_qty_uom': consumed_qty,
                 'product_expected_qty_uom': expected_qty
             }))
@@ -1790,7 +1790,7 @@ class MrpProduction(models.Model):
         if self.env.context.get('skip_backorder', False):
             return quantity_issues
         for order in self:
-            if not order.product_uom_id.is_zero(order._get_quantity_to_backorder()):
+            if not order.uom_id.is_zero(order._get_quantity_to_backorder()):
                 quantity_issues.append(order)
         return quantity_issues
 
@@ -1896,7 +1896,7 @@ class MrpProduction(models.Model):
             for move in finish_moves:
                 if move.has_tracking != 'none' and not move.lot_ids:
                     move.lot_ids = order.lot_producing_ids.ids
-                move.quantity = order.product_uom_id.round(order.qty_producing - order.qty_produced, rounding_method='HALF-UP')
+                move.quantity = order.uom_id.round(order.qty_producing - order.qty_produced, rounding_method='HALF-UP')
                 extra_vals = order._prepare_finished_extra_vals()
                 if extra_vals:
                     move.move_line_ids.write(extra_vals)
@@ -1970,7 +1970,7 @@ class MrpProduction(models.Model):
                 amounts[production] = _default_amounts(production)
                 continue
             total_amount = sum(mo_amounts)
-            diff = production.product_uom_id.compare(production.product_qty, total_amount)
+            diff = production.uom_id.compare(production.product_qty, total_amount)
             if diff > 0 and not cancel_remaining_qty:
                 amounts[production].append(production.product_qty - total_amount)
                 has_backorder_to_ignore[production] = True
@@ -2079,7 +2079,7 @@ class MrpProduction(models.Model):
             product_uom = initial_move.product_id.uom_id
             if not initial_move.picked:
                 for move_line in initial_move.move_line_ids:
-                    available_qty = move_line.product_uom_id._compute_quantity(move_line.quantity, product_uom, rounding_method="HALF-UP")
+                    available_qty = move_line.uom_id._compute_quantity(move_line.quantity, product_uom, rounding_method="HALF-UP")
                     if product_uom.compare(available_qty, 0) <= 0:
                         continue
                     ml_by_move.append((available_qty, move_line, move_line.copy_data()[0]))
@@ -2091,8 +2091,8 @@ class MrpProduction(models.Model):
 
             for index, (quantity, move_line, ml_vals) in enumerate(ml_by_move):
                 taken_qty = min(quantity, move_qty_to_reserve)
-                taken_qty_uom = product_uom._compute_quantity(taken_qty, move_line.product_uom_id, rounding_method="HALF-UP")
-                if move_line.product_uom_id.is_zero(taken_qty_uom):
+                taken_qty_uom = product_uom._compute_quantity(taken_qty, move_line.uom_id, rounding_method="HALF-UP")
+                if move_line.uom_id.is_zero(taken_qty_uom):
                     continue
                 move_line.write({
                     'quantity': taken_qty_uom,
@@ -2101,7 +2101,7 @@ class MrpProduction(models.Model):
                 move_qty_to_reserve -= taken_qty
                 ml_by_move[index] = (quantity - taken_qty, move_line, ml_vals)
 
-                if move.product_uom.compare(move_qty_to_reserve, 0) <= 0:
+                if move.uom_id.compare(move_qty_to_reserve, 0) <= 0:
                     assigned_moves.add(move.id)
                     move = moves and moves.pop(0)
                     move_qty_to_reserve = move and move.product_qty or 0
@@ -2110,10 +2110,10 @@ class MrpProduction(models.Model):
                 while product_uom.compare(quantity, 0) > 0 and move:
                     # Do not create `stock.move.line` if there is no initial demand on `stock.move`
                     taken_qty = min(move_qty_to_reserve, quantity)
-                    taken_qty_uom = product_uom._compute_quantity(taken_qty, move_line.product_uom_id, rounding_method="HALF-UP")
+                    taken_qty_uom = product_uom._compute_quantity(taken_qty, move_line.uom_id, rounding_method="HALF-UP")
                     if move == initial_move:
                         move_line.quantity += taken_qty_uom
-                    elif not move_line.product_uom_id.is_zero(taken_qty_uom):
+                    elif not move_line.uom_id.is_zero(taken_qty_uom):
                         new_ml_vals = dict(
                             ml_vals,
                             quantity=taken_qty_uom,
@@ -2123,7 +2123,7 @@ class MrpProduction(models.Model):
                     quantity -= taken_qty
                     move_qty_to_reserve -= taken_qty
 
-                    if move.product_uom.compare(move_qty_to_reserve, 0) <= 0:
+                    if move.uom_id.compare(move_qty_to_reserve, 0) <= 0:
                         assigned_moves.add(move.id)
                         move = moves and moves.pop(0)
                         move_qty_to_reserve = move and move.product_qty or 0
@@ -2315,7 +2315,7 @@ class MrpProduction(models.Model):
         self._button_mark_done_sanity_checks()
         production_auto_ids = set()
         for production in self:
-            if not production.product_uom_id.is_zero(production.qty_producing):
+            if not production.uom_id.is_zero(production.qty_producing):
                 production.move_raw_ids.filtered(
                     lambda move: move.manual_consumption and not move.picked
                 ).picked = True
@@ -2524,7 +2524,7 @@ class MrpProduction(models.Model):
             'bom_id': bom_id.id,
             'picking_type_id': self.picking_type_id.id,
             'product_qty': sum(production.product_uom_qty for production in self),
-            'product_uom_id': product_id.uom_id.id,
+            'uom_id': product_id.uom_id.id,
             'location_final_id': all(mo.location_final_id for mo in self) and len(self.location_final_id) == 1 and self.location_final_id.id,
             'user_id': user_id.id,
             'reference_ids': [Command.link(r.id) for r in self.reference_ids],
@@ -2577,7 +2577,7 @@ class MrpProduction(models.Model):
         """
         self.ensure_one()
         product_qty = self.product_qty
-        uom = self.product_uom_id
+        uom = self.uom_id
         moves_to_unlink = self.env['stock.move']
         workorders_to_unlink = self.env['mrp.workorder']
         # For draft MO, all the work will be done by compute methods.
@@ -2598,7 +2598,7 @@ class MrpProduction(models.Model):
             if self.state == 'draft':
                 # we reset the product_qty/uom when the bom is changed on a draft MO
                 # change them back to the original value
-                self.write({'product_qty': product_qty, 'product_uom_id': uom.id})
+                self.write({'product_qty': product_qty, 'uom_id': uom.id})
             return
 
         def operation_key_values(record):
@@ -2644,7 +2644,7 @@ class MrpProduction(models.Model):
             workorder_vals = {
                 'name': operation.name,
                 'operation_id': operation.id,
-                'product_uom_id': self.product_uom_id.id,
+                'uom_id': self.uom_id.id,
                 'production_id': self.id,
                 'state': 'blocked',
                 'workcenter_id': operation.workcenter_id.id,
@@ -2662,8 +2662,8 @@ class MrpProduction(models.Model):
                         bom_line, bom_qty = bom_lines_by_id.pop((_bom_line.id, move_raw.product_id.id))
                         if bom_line:
                             break
-            move_raw_qty = bom_line and move_raw.product_uom._compute_quantity(
-                move_raw.product_uom_qty * ratio, bom_line.product_uom_id
+            move_raw_qty = bom_line and move_raw.uom_id._compute_quantity(
+                move_raw.product_uom_qty * ratio, bom_line.uom_id
             )
             if bom_line and (
                     not move_raw.bom_line_id or
@@ -2674,7 +2674,7 @@ class MrpProduction(models.Model):
                 move_raw.bom_line_id = bom_line
                 move_raw.product_id = bom_line.product_id
                 move_raw.product_uom_qty = bom_qty / ratio
-                move_raw.product_uom = bom_line.product_uom_id
+                move_raw.uom_id = bom_line.uom_id
                 if move_raw.operation_id != bom_line.operation_id:
                     move_raw.operation_id = bom_line.operation_id
                     move_raw.workorder_id = self.workorder_ids.filtered(lambda wo: wo.operation_id == move_raw.operation_id)
@@ -2686,7 +2686,7 @@ class MrpProduction(models.Model):
             raw_move_vals = self._get_move_raw_values(
                 bom_line.product_id,
                 bom_qty / ratio,
-                bom_line.product_uom_id,
+                bom_line.uom_id,
                 bom_line=bom_line
             )
             raw_moves_values.append(raw_move_vals)
@@ -2700,8 +2700,8 @@ class MrpProduction(models.Model):
                     if move_byproduct.product_id == _bom_byproduct.product_id:
                         bom_byproduct = bom_byproducts_by_id.pop(_bom_byproduct.id)
                         break
-            move_byproduct_qty = bom_byproduct and move_byproduct.product_uom._compute_quantity(
-                move_byproduct.product_uom_qty * ratio, bom_byproduct.product_uom_id
+            move_byproduct_qty = bom_byproduct and move_byproduct.uom_id._compute_quantity(
+                move_byproduct.product_uom_qty * ratio, bom_byproduct.uom_id
             )
             if bom_byproduct and (
                     not move_byproduct.byproduct_id or
@@ -2711,7 +2711,7 @@ class MrpProduction(models.Model):
                 move_byproduct.byproduct_id = bom_byproduct
                 move_byproduct.cost_share = bom_byproduct.cost_share
                 move_byproduct.product_uom_qty = bom_byproduct.product_qty / ratio
-                move_byproduct.product_uom = bom_byproduct.product_uom_id
+                move_byproduct.uom_id = bom_byproduct.uom_id
             elif not bom_byproduct:
                 moves_to_unlink |= move_byproduct
         # For each remaining BoM's by-product, creates a move finished.
@@ -2719,7 +2719,7 @@ class MrpProduction(models.Model):
         for bom_byproduct in bom_byproducts_by_id.values():
             qty = bom_byproduct.product_qty / ratio
             move_byproduct_vals = self._get_move_finished_values(
-                bom_byproduct.product_id.id, qty, bom_byproduct.product_uom_id.id,
+                bom_byproduct.product_id.id, qty, bom_byproduct.uom_id.id,
                 bom_byproduct.operation_id.id, bom_byproduct.id, bom_byproduct.cost_share
             )
             byproduct_values.append(move_byproduct_vals)
@@ -2739,7 +2739,7 @@ class MrpProduction(models.Model):
     def _get_ratio_between_mo_and_bom_quantities(self, bom):
         self.ensure_one()
         bom_product_uom = (bom.product_id or bom.product_tmpl_id).uom_id
-        bom_qty = bom.product_uom_id._compute_quantity(bom.product_qty, bom_product_uom)
+        bom_qty = bom.uom_id._compute_quantity(bom.product_qty, bom_product_uom)
         ratio = bom_qty / self.product_uom_qty
         return ratio
 
@@ -2755,7 +2755,7 @@ class MrpProduction(models.Model):
             if move.has_tracking != 'serial' or move.product_id == self.product_id:
                 continue
             for move_line in move.move_line_ids:
-                if move_line.product_uom_id.is_zero(move_line.quantity):
+                if move_line.uom_id.is_zero(move_line.quantity):
                     continue
                 if self._are_finished_serials_already_produced(move_line.lot_id, excluded_sml=move_line):
                     raise UserError(_('The serial number %(number)s used for byproduct %(product_name)s has already been produced',
@@ -2767,7 +2767,7 @@ class MrpProduction(models.Model):
             if move.has_tracking != 'serial' or not move.picked:
                 continue
             for move_line in move.move_line_ids:
-                if not move_line.picked or move_line.product_uom_id.is_zero(move_line.quantity) or not move_line.lot_id:
+                if not move_line.picked or move_line.uom_id.is_zero(move_line.quantity) or not move_line.lot_id:
                     continue
                 sml_sn = move_line.lot_id
                 message = _('The serial number %(number)s used for component %(component)s has already been consumed',

--- a/addons/mrp/models/mrp_routing.py
+++ b/addons/mrp/models/mrp_routing.py
@@ -94,7 +94,7 @@ class MrpRoutingWorkcenter(models.Model):
             cycle_number = 0  # Never 0 unless infinite item['workcenter_id'].capacity
             for item in data:
                 total_duration += item['duration']
-                (capacity, _setup, _cleanup) = item['workcenter_id']._get_capacity(item.product_id, item.product_uom_id, operation.bom_id.product_qty or 1)
+                (capacity, _setup, _cleanup) = item['workcenter_id']._get_capacity(item.product_id, item.uom_id, operation.bom_id.product_qty or 1)
                 cycle_number += float_round((item['qty_produced'] / capacity), precision_digits=0, rounding_method='UP')
             if cycle_number:
                 operation.time_cycle = total_duration / cycle_number
@@ -110,7 +110,7 @@ class MrpRoutingWorkcenter(models.Model):
                 operation.show_time_total = False
                 continue
             quantity = self.env.context.get('quantity', operation.bom_id.product_qty or 1)
-            unit = self.env.context.get('unit', operation.bom_id.product_uom_id)
+            unit = self.env.context.get('unit', operation.bom_id.uom_id)
             (capacity, setup, cleanup) = workcenter._get_capacity(product, unit, operation.bom_id.product_qty or 1)
             operation.cycle_number = float_round(quantity / capacity, precision_digits=0, rounding_method="UP")
             operation.time_total = setup + cleanup + operation.cycle_number * operation.time_cycle * 100.0 / (workcenter.time_efficiency or 100.0)

--- a/addons/mrp/models/mrp_unbuild.py
+++ b/addons/mrp/models/mrp_unbuild.py
@@ -29,9 +29,9 @@ class MrpUnbuild(models.Model):
         digits='Product Unit',
         compute='_compute_product_qty', store=True, precompute=True, readonly=False,
         required=True)
-    product_uom_id = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', 'Unit',
-        compute='_compute_product_uom_id', store=True, readonly=False, precompute=True,
+        compute='_compute_uom_id', store=True, readonly=False, precompute=True,
         required=True)
     bom_id = fields.Many2one(
         'mrp.bom', 'Bill of Material',
@@ -86,12 +86,12 @@ class MrpUnbuild(models.Model):
     )
 
     @api.depends('mo_id', 'product_id')
-    def _compute_product_uom_id(self):
+    def _compute_uom_id(self):
         for record in self:
             if record.mo_id.product_id and record.mo_id.product_id == record.product_id:
-                record.product_uom_id = record.mo_id.product_uom_id.id
+                record.uom_id = record.mo_id.uom_id.id
             else:
-                record.product_uom_id = record.product_id.uom_id.id
+                record.uom_id = record.product_id.uom_id.id
 
     @api.depends('company_id')
     def _compute_location_id(self):
@@ -146,7 +146,7 @@ class MrpUnbuild(models.Model):
             'lot_id': self.lot_id.id,
             'quantity': finished_move.product_uom_qty - finished_move.quantity,
             'product_id': finished_move.product_id.id,
-            'product_uom_id': finished_move.product_uom.id,
+            'uom_id': finished_move.uom_id.id,
             'location_id': finished_move.location_id.id,
             'location_dest_id': finished_move.location_dest_id.id,
         }
@@ -157,7 +157,7 @@ class MrpUnbuild(models.Model):
             'lot_id': origin_move_line.lot_id.id,
             'quantity': taken_quantity,
             'product_id': move.product_id.id,
-            'product_uom_id': origin_move_line.product_uom_id.id,
+            'uom_id': origin_move_line.uom_id.id,
             'location_id': move.location_id.id,
             'location_dest_id': move.location_dest_id.id,
         }
@@ -196,19 +196,19 @@ class MrpUnbuild(models.Model):
             raise UserError(error_message)
 
         for finished_move in finished_moves:
-            if float_compare(finished_move.product_uom_qty, finished_move.quantity, precision_rounding=finished_move.product_uom.rounding) > 0:
+            if float_compare(finished_move.product_uom_qty, finished_move.quantity, precision_rounding=finished_move.uom_id.rounding) > 0:
                 finished_move_line_vals = self._prepare_finished_move_line_vals(finished_move)
                 self.env['stock.move.line'].create(finished_move_line_vals)
 
         # TODO: Will fail if user do more than one unbuild with lot on the same MO. Need to check what other unbuild has aready took
         qty_already_used = defaultdict(float)
         for move in produce_moves | consume_moves:
-            if float_compare(move.product_uom_qty, move.quantity, precision_rounding=move.product_uom.rounding) < 1:
+            if float_compare(move.product_uom_qty, move.quantity, precision_rounding=move.uom_id.rounding) < 1:
                 continue
             original_move = move in produce_moves and self.mo_id.move_raw_ids or self.mo_id.move_finished_ids
             original_move = original_move.filtered(lambda m: m.product_id == move.product_id)
             if not original_move:
-                move.quantity = move.product_uom.round(move.product_uom_qty)
+                move.quantity = move.uom_id.round(move.product_uom_qty)
                 continue
             needed_quantity = move.product_uom_qty
             moves_lines = original_move.mapped('move_line_ids')
@@ -219,7 +219,7 @@ class MrpUnbuild(models.Model):
             for move_line in moves_lines:
                 # Iterate over all move_lines until we unbuilded the correct quantity.
                 taken_quantity = min(needed_quantity, move_line.quantity - qty_already_used[move_line])
-                taken_quantity = move.product_uom.round(taken_quantity)
+                taken_quantity = move.uom_id.round(taken_quantity)
                 if taken_quantity:
                     move_line_vals = self._prepare_move_line_vals(move, move_line, taken_quantity)
                     if move_line.owner_id:
@@ -238,7 +238,7 @@ class MrpUnbuild(models.Model):
         if self.mo_id:
             unbuild_msg = _("%(qty)s %(measure)s unbuilt in %(order)s",
                 qty=self.product_qty,
-                measure=self.product_uom_id.name,
+                measure=self.uom_id.name,
                 order=self._get_html_link(),
             )
             self.mo_id.message_post(
@@ -252,17 +252,17 @@ class MrpUnbuild(models.Model):
         for unbuild in self:
             if unbuild.mo_id:
                 finished_moves = unbuild.mo_id.move_finished_ids.filtered(lambda move: move.state == 'done')
-                factor = unbuild.product_qty / unbuild.mo_id.product_uom_id._compute_quantity(unbuild.mo_id.qty_produced, unbuild.product_uom_id)
+                factor = unbuild.product_qty / unbuild.mo_id.uom_id._compute_quantity(unbuild.mo_id.qty_produced, unbuild.uom_id)
                 for finished_move in finished_moves:
                     moves += unbuild._generate_move_from_existing_move(finished_move, factor, unbuild.location_id, finished_move.location_id)
             else:
-                factor = unbuild.product_uom_id._compute_quantity(unbuild.product_qty, unbuild.bom_id.product_uom_id) / unbuild.bom_id.product_qty
-                moves += unbuild._generate_move_from_bom_line(self.product_id, self.product_uom_id, unbuild.product_qty)
+                factor = unbuild.uom_id._compute_quantity(unbuild.product_qty, unbuild.bom_id.uom_id) / unbuild.bom_id.product_qty
+                moves += unbuild._generate_move_from_bom_line(self.product_id, self.uom_id, unbuild.product_qty)
                 for byproduct in unbuild.bom_id.byproduct_ids:
                     if byproduct._skip_byproduct_line(unbuild.product_id):
                         continue
                     quantity = byproduct.product_qty * factor
-                    moves += unbuild._generate_move_from_bom_line(byproduct.product_id, byproduct.product_uom_id, quantity, byproduct_id=byproduct.id)
+                    moves += unbuild._generate_move_from_bom_line(byproduct.product_id, byproduct.uom_id, quantity, byproduct_id=byproduct.id)
         return moves
 
     def _generate_produce_moves(self):
@@ -270,14 +270,14 @@ class MrpUnbuild(models.Model):
         for unbuild in self:
             if unbuild.mo_id:
                 raw_moves = unbuild.mo_id.move_raw_ids.filtered(lambda move: move.state == 'done')
-                factor = unbuild.product_qty / unbuild.mo_id.product_uom_id._compute_quantity(unbuild.mo_id.qty_produced, unbuild.product_uom_id)
+                factor = unbuild.product_qty / unbuild.mo_id.uom_id._compute_quantity(unbuild.mo_id.qty_produced, unbuild.uom_id)
                 for raw_move in raw_moves:
                     moves += unbuild._generate_move_from_existing_move(raw_move, factor, raw_move.location_dest_id, self.location_dest_id)
             else:
-                factor = unbuild.product_uom_id._compute_quantity(unbuild.product_qty, unbuild.bom_id.product_uom_id) / unbuild.bom_id.product_qty
+                factor = unbuild.uom_id._compute_quantity(unbuild.product_qty, unbuild.bom_id.uom_id) / unbuild.bom_id.product_qty
                 boms, lines = unbuild.bom_id.explode(unbuild.product_id, factor, picking_type=unbuild.bom_id.picking_type_id)
                 for line, line_data in lines:
-                    moves += unbuild._generate_move_from_bom_line(line.product_id, line.product_uom_id, line_data['qty'], bom_line_id=line.id)
+                    moves += unbuild._generate_move_from_bom_line(line.product_id, line.uom_id, line_data['qty'], bom_line_id=line.id)
         return moves
 
     def _generate_move_from_existing_move(self, move, factor, location_id, location_dest_id):
@@ -285,7 +285,7 @@ class MrpUnbuild(models.Model):
             'date': self.create_date,
             'product_id': move.product_id.id,
             'product_uom_qty': move.quantity * factor,
-            'product_uom': move.product_uom.id,
+            'uom_id': move.uom_id.id,
             'procure_method': 'make_to_stock',
             'location_dest_id': location_dest_id.id,
             'location_id': location_id.id,
@@ -306,7 +306,7 @@ class MrpUnbuild(models.Model):
             'byproduct_id': byproduct_id,
             'product_id': product.id,
             'product_uom_qty': quantity,
-            'product_uom': product_uom.id,
+            'uom_id': product_uom.id,
             'procure_method': 'make_to_stock',
             'location_dest_id': location_dest_id.id,
             'location_id': location_id.id,
@@ -319,7 +319,7 @@ class MrpUnbuild(models.Model):
         self.ensure_one()
         precision = self.env['decimal.precision'].precision_get('Product Unit')
         available_qty = self.env['stock.quant']._get_available_quantity(self.product_id, self.location_id, self.lot_id, strict=True)
-        unbuild_qty = self.product_uom_id._compute_quantity(self.product_qty, self.product_id.uom_id)
+        unbuild_qty = self.uom_id._compute_quantity(self.product_qty, self.product_id.uom_id)
         if float_compare(available_qty, unbuild_qty, precision_digits=precision) >= 0:
             return self.action_unbuild()
         else:

--- a/addons/mrp/models/mrp_workcenter.py
+++ b/addons/mrp/models/mrp_workcenter.py
@@ -425,14 +425,14 @@ class MrpWorkcenter(models.Model):
 
     def _get_capacity(self, product, unit, default_capacity=1):
         capacity = self.capacity_ids.sorted(lambda c: (
-            not (c.product_id == product and c.product_uom_id == product.uom_id),
-            not (not c.product_id and c.product_uom_id == unit),
-            not (not c.product_id and c.product_uom_id == product.uom_id),
+            not (c.product_id == product and c.uom_id == product.uom_id),
+            not (not c.product_id and c.uom_id == unit),
+            not (not c.product_id and c.uom_id == product.uom_id),
         ))[:1]
-        if capacity and capacity.product_id in [product, self.env['product.product']] and capacity.product_uom_id in [product.uom_id, unit]:
+        if capacity and capacity.product_id in [product, self.env['product.product']] and capacity.uom_id in [product.uom_id, unit]:
             if float_is_zero(capacity.capacity, 0):
                 return (default_capacity, capacity.time_start, capacity.time_stop)
-            return (capacity.product_uom_id._compute_quantity(capacity.capacity, unit), capacity.time_start, capacity.time_stop)
+            return (capacity.uom_id._compute_quantity(capacity.capacity, unit), capacity.time_start, capacity.time_stop)
         return (default_capacity, self.time_start, self.time_stop)
 
 
@@ -624,8 +624,8 @@ class MrpWorkcenterCapacity(models.Model):
 
     workcenter_id = fields.Many2one('mrp.workcenter', string='Work Center', required=True, index=True)
     product_id = fields.Many2one('product.product', string='Product')
-    product_uom_id = fields.Many2one('uom.uom', string='Unit',
-        compute="_compute_product_uom_id", precompute=True, store=True, readonly=False, required=True)
+    uom_id = fields.Many2one('uom.uom', string='Unit',
+        compute="_compute_uom_id", precompute=True, store=True, readonly=False, required=True)
     capacity = fields.Float('Capacity', help="Number of pieces that can be produced in parallel for this product or for all, depending on the unit.")
     time_start = fields.Float('Setup Time (minutes)', default=_default_time_start, help="Time in minutes for the setup.")
     time_stop = fields.Float('Cleanup Time (minutes)', default=_default_time_stop, help="Time in minutes for the cleaning.")
@@ -635,11 +635,11 @@ class MrpWorkcenterCapacity(models.Model):
         'Capacity should be a non-negative number.',
     )
     _workcenter_product_product_uom_unique = models.UniqueIndex(
-        '(workcenter_id, COALESCE(product_id, 0), product_uom_id)',
+        '(workcenter_id, COALESCE(product_id, 0), uom_id)',
         'Product/Unit capacity should be unique for each workcenter.'
     )
 
     @api.depends('product_id')
-    def _compute_product_uom_id(self):
+    def _compute_uom_id(self):
         for capacity in self:
-            capacity.product_uom_id = capacity.product_id.uom_id or self.env.ref('uom.product_uom_unit')
+            capacity.uom_id = capacity.product_id.uom_id or self.env.ref('uom.product_uom_unit')

--- a/addons/mrp/models/mrp_workorder.py
+++ b/addons/mrp/models/mrp_workorder.py
@@ -39,7 +39,7 @@ class MrpWorkorder(models.Model):
         string='Workcenter Status', related='workcenter_id.working_state') # technical: used in views only
     product_id = fields.Many2one(related='production_id.product_id')
     product_tracking = fields.Selection(related="product_id.tracking")
-    product_uom_id = fields.Many2one(related='production_id.product_uom_id')
+    uom_id = fields.Many2one(related='production_id.uom_id')
     product_variant_attributes = fields.Many2many('product.template.attribute.value', related='product_id.product_template_attribute_value_ids')
     production_id = fields.Many2one('mrp.production', 'Manufacturing Order', required=True, check_company=True, readonly=True, index='btree')
     production_availability = fields.Selection(
@@ -149,9 +149,9 @@ class MrpWorkorder(models.Model):
     @api.depends('qty_ready')
     def _compute_state(self):
         for workorder in self:
-            if not workorder.product_uom_id or workorder.state not in ('blocked', 'ready'):
+            if not workorder.uom_id or workorder.state not in ('blocked', 'ready'):
                 continue
-            has_qty_ready = workorder.product_uom_id.compare(workorder.qty_ready, 0) > 0
+            has_qty_ready = workorder.uom_id.compare(workorder.qty_ready, 0) > 0
             if has_qty_ready:
                 workorder.write({'state': 'ready'})
             else:
@@ -325,11 +325,11 @@ class MrpWorkorder(models.Model):
         mo_dirty.workorder_ids._action_confirm()
         return res
 
-    @api.depends('production_id.product_qty', 'qty_produced', 'production_id.product_uom_id')
+    @api.depends('production_id.product_qty', 'qty_produced', 'production_id.uom_id')
     def _compute_is_produced(self):
         self.is_produced = False
-        for order in self.filtered(lambda p: p.production_id and p.production_id.product_uom_id):
-            order.is_produced = order.production_id.product_uom_id.compare(order.qty_produced, order.qty_production) >= 0
+        for order in self.filtered(lambda p: p.production_id and p.production_id.uom_id):
+            order.is_produced = order.production_id.uom_id.compare(order.qty_produced, order.qty_production) >= 0
 
     @api.depends('operation_id', 'workcenter_id', 'qty_producing', 'qty_production')
     def _compute_duration_expected(self):
@@ -476,7 +476,7 @@ class MrpWorkorder(models.Model):
             for wo in self:
                 if wo.state in ['done', 'cancel']:
                     raise UserError(_('You cannot change the quantity produced of a work order that is in done or cancel state.'))
-                elif wo.product_uom_id.compare(values['qty_produced'], 0) < 0:
+                elif wo.uom_id.compare(values['qty_produced'], 0) < 0:
                     raise UserError(_('The quantity produced must be positive.'))
 
         if 'production_id' in values and any(values['production_id'] != w.production_id.id for w in self):
@@ -521,12 +521,12 @@ class MrpWorkorder(models.Model):
 
         res = super().write(values)
         productions = self.production_id.filtered(
-            lambda p: p.product_uom_id.compare(values.get('qty_produced', 0), 0) > 0
+            lambda p: p.uom_id.compare(values.get('qty_produced', 0), 0) > 0
         )
         if 'qty_produced' in values and productions:
             for production in productions:
                 min_wo_qty = min(production.workorder_ids.mapped('qty_produced'))
-                if production.product_uom_id.compare(min_wo_qty, 0) > 0:
+                if production.uom_id.compare(min_wo_qty, 0) > 0:
                     production.workorder_ids.filtered(lambda w: w.state != 'done').qty_producing = min_wo_qty
             self._set_qty_producing()
 
@@ -693,11 +693,11 @@ class MrpWorkorder(models.Model):
 
         for move in moves_to_pick:
             production_id = move.raw_material_production_id or move.production_id
-            if production_id.product_uom_id.is_zero(production_id.qty_producing):
+            if production_id.uom_id.is_zero(production_id.qty_producing):
                 qty_available = production_id.product_qty
             else:
                 qty_available = production_id.qty_producing
-            new_qty = move.product_uom.round(qty_available * move.unit_factor)
+            new_qty = move.uom_id.round(qty_available * move.unit_factor)
             move._set_quantity_done(new_qty)
 
         moves_to_pick.picked = True
@@ -760,11 +760,11 @@ class MrpWorkorder(models.Model):
         action['res_id'] = self.id
         return action
 
-    @api.depends('qty_production', 'qty_reported_from_previous_wo', 'qty_produced', 'production_id.product_uom_id')
+    @api.depends('qty_production', 'qty_reported_from_previous_wo', 'qty_produced', 'production_id.uom_id')
     def _compute_qty_remaining(self):
         for wo in self:
-            if wo.production_id.product_uom_id:
-                wo.qty_remaining = max(wo.production_id.product_uom_id.round(wo.qty_production - wo.qty_reported_from_previous_wo - wo.qty_produced), 0)
+            if wo.production_id.uom_id:
+                wo.qty_remaining = max(wo.production_id.uom_id.round(wo.qty_production - wo.qty_reported_from_previous_wo - wo.qty_produced), 0)
             else:
                 wo.qty_remaining = 0
 
@@ -772,7 +772,7 @@ class MrpWorkorder(models.Model):
         self.ensure_one()
         if not self.workcenter_id:
             return self.duration_expected
-        capacity, setup, cleanup = self.workcenter_id._get_capacity(self.product_id, self.product_uom_id, self.production_bom_id.product_qty or 1)
+        capacity, setup, cleanup = self.workcenter_id._get_capacity(self.product_id, self.uom_id, self.production_bom_id.product_qty or 1)
         if not self.operation_id:
             duration_expected_working = (self.duration_expected - setup - cleanup) * self.workcenter_id.time_efficiency / 100.0
             if duration_expected_working < 0:
@@ -789,7 +789,7 @@ class MrpWorkorder(models.Model):
             duration_expected_working = (self.duration_expected - setup - cleanup) * self.workcenter_id.time_efficiency / (100.0 * cycle_number)
             if duration_expected_working < 0:
                 duration_expected_working = 0
-            capacity, setup, cleanup = alternative_workcenter._get_capacity(self.product_id, self.product_uom_id, self.production_bom_id.product_qty or 1)
+            capacity, setup, cleanup = alternative_workcenter._get_capacity(self.product_id, self.uom_id, self.production_bom_id.product_qty or 1)
             cycle_number = float_round(qty_production / capacity, precision_digits=0, rounding_method='UP')
             return setup + cleanup + cycle_number * duration_expected_working * 100.0 / alternative_workcenter.time_efficiency
         time_cycle = self.operation_id.time_cycle

--- a/addons/mrp/models/product.py
+++ b/addons/mrp/models/product.py
@@ -304,13 +304,13 @@ class ProductProduct(models.Model):
                 component = component.with_context(mrp_compute_quantities=qties).with_prefetch(prefetch_component_ids)
                 qty_per_kit = 0
                 for bom_line, bom_line_data in bom_sub_lines:
-                    if not component.is_storable or bom_line.product_uom_id.is_zero(bom_line_data['qty']):
+                    if not component.is_storable or bom_line.uom_id.is_zero(bom_line_data['qty']):
                         # As BoMs allow components with 0 qty, a.k.a. optionnal components, we simply skip those
                         # to avoid a division by zero. The same logic is applied to non-storable products as those
                         # products have 0 qty available.
                         continue
                     uom_qty_per_kit = bom_line_data['qty'] / bom_line_data['original_qty']
-                    qty_per_kit += bom_line.product_uom_id._compute_quantity(uom_qty_per_kit, bom_line.product_id.uom_id, round=False, raise_if_failure=False)
+                    qty_per_kit += bom_line.uom_id._compute_quantity(uom_qty_per_kit, bom_line.product_id.uom_id, round=False, raise_if_failure=False)
                 if not qty_per_kit:
                     continue
                 component_res = (
@@ -440,7 +440,7 @@ class ProductProduct(models.Model):
     def _update_uom(self, to_uom_id):
         for uom, product_template, boms in self.env['mrp.bom']._read_group(
             [('product_tmpl_id', 'in', self.product_tmpl_id.ids)],
-            ['product_uom_id', 'product_tmpl_id'],
+            ['uom_id', 'product_tmpl_id'],
             ['id:recordset'],
         ):
             if product_template.uom_id != uom:
@@ -448,11 +448,11 @@ class ProductProduct(models.Model):
                 'than %(uom)s have already been used for this product, the change of unit of measure can not be done.'
                 'If you want to change it, please archive the product and create a new one.',
                 problem_uom=uom.name, uom=product_template.uom_id.name))
-            boms.product_uom_id = to_uom_id
+            boms.uom_id = to_uom_id
 
         for uom, product, bom_lines in self.env['mrp.bom.line']._read_group(
             [('product_id', 'in', self.ids)],
-            ['product_uom_id', 'product_id'],
+            ['uom_id', 'product_id'],
             ['id:recordset'],
         ):
             if product.product_tmpl_id.uom_id != uom:
@@ -460,11 +460,11 @@ class ProductProduct(models.Model):
                 'than %(uom)s have already been used for this product, the change of unit of measure can not be done.'
                 'If you want to change it, please archive the product and create a new one.',
                 problem_uom=uom.name, uom=product.product_tmpl_id.uom_id.name))
-            bom_lines.product_uom_id = to_uom_id
+            bom_lines.uom_id = to_uom_id
 
         for uom, product, productions in self.env['mrp.production']._read_group(
             [('product_id', 'in', self.ids)],
-            ['product_uom_id', 'product_id'],
+            ['uom_id', 'product_id'],
             ['id:recordset'],
         ):
             if product.product_tmpl_id.uom_id != uom:
@@ -472,6 +472,6 @@ class ProductProduct(models.Model):
                 'than %(uom)s have already been used for this product, the change of unit of measure can not be done.'
                 'If you want to change it, please archive the product and create a new one.',
                 problem_uom=uom.name, uom=product.product_tmpl_id.uom_id.name))
-            productions.product_uom_id = to_uom_id
+            productions.uom_id = to_uom_id
 
         return super()._update_uom(to_uom_id)

--- a/addons/mrp/models/stock_move.py
+++ b/addons/mrp/models/stock_move.py
@@ -63,18 +63,18 @@ class StockMove(models.Model):
         help="When activated, then the registration of consumption for that component is recorded manually exclusively.\n"
              "If not activated, and any of the components consumption is edited manually on the manufacturing order, Odoo assumes manual consumption also.")
 
-    @api.depends('product_id.bom_ids', 'product_id.bom_ids.product_uom_id')
+    @api.depends('product_id.bom_ids', 'product_id.bom_ids.uom_id')
     def _compute_allowed_uom_ids(self):
         super()._compute_allowed_uom_ids()
         for move in self:
-            move.allowed_uom_ids |= move.product_id.bom_ids.product_uom_id
+            move.allowed_uom_ids |= move.product_id.bom_ids.uom_id
 
     @api.depends('production_id')
     def _compute_packaging_uom_id(self):
         super()._compute_packaging_uom_id()
         for move in self:
             if move.production_id:
-                move.packaging_uom_id = move.production_id.product_uom_id
+                move.packaging_uom_id = move.production_id.uom_id
 
     @api.depends('product_id')
     def _compute_manual_consumption(self):
@@ -186,14 +186,14 @@ class StockMove(models.Model):
             if production:
                 move.reference_ids = [Command.set(production.reference_ids.ids)]
 
-    @api.depends('raw_material_production_id.qty_producing', 'product_uom_qty', 'product_uom')
+    @api.depends('raw_material_production_id.qty_producing', 'product_uom_qty', 'uom_id')
     def _compute_should_consume_qty(self):
         for move in self:
             mo = move.raw_material_production_id
-            if not mo or not move.product_uom:
+            if not mo or not move.uom_id:
                 move.should_consume_qty = 0
                 continue
-            move.should_consume_qty = move.product_uom.round((mo.qty_producing - mo.qty_produced) * move.unit_factor)
+            move.should_consume_qty = move.uom_id.round((mo.qty_producing - mo.qty_produced) * move.unit_factor)
 
     @api.depends('byproduct_id')
     def _compute_show_info(self):
@@ -212,25 +212,25 @@ class StockMove(models.Model):
                 move.display_import_lot = False
                 move.display_assign_serial = False
 
-    @api.onchange('product_uom_qty', 'product_uom')
+    @api.onchange('product_uom_qty', 'uom_id')
     def _onchange_product_uom_qty(self):
-        if self.product_uom and self.raw_material_production_id and self.has_tracking == 'none'\
+        if self.uom_id and self.raw_material_production_id and self.has_tracking == 'none'\
             and self.state not in ('draft', 'cancel', 'done'):
             mo = self.raw_material_production_id
-            new_qty = self.product_uom.round((mo.qty_producing - mo.qty_produced) * self.unit_factor)
+            new_qty = self.uom_id.round((mo.qty_producing - mo.qty_produced) * self.unit_factor)
             self.quantity = new_qty
 
-    @api.onchange('quantity', 'product_uom', 'picked')
+    @api.onchange('quantity', 'uom_id', 'picked')
     def _onchange_quantity(self):
-        if self.raw_material_production_id and self.product_uom and \
-            not float_is_zero(self.quantity, precision_rounding=self.product_uom.rounding) and self.product_uom.compare(self.product_uom_qty, self.quantity) != 0:
+        if self.raw_material_production_id and self.uom_id and \
+            not float_is_zero(self.quantity, precision_rounding=self.uom_id.rounding) and self.uom_id.compare(self.product_uom_qty, self.quantity) != 0:
             self.manual_consumption = True
             self.picked = True
 
     @api.constrains('quantity', 'raw_material_production_id')
     def _check_negative_quantity(self):
         for move in self:
-            if move.raw_material_production_id and move.product_uom.compare(move.quantity, 0) < 0:
+            if move.raw_material_production_id and move.uom_id.compare(move.quantity, 0) < 0:
                 raise ValidationError(_("Please enter a positive quantity."))
 
     @api.model_create_multi
@@ -316,11 +316,11 @@ class StockMove(models.Model):
         to_assign = self.env['stock.move']
         self._adjust_procure_method()
         for move in self:
-            if move.product_uom.compare(move.product_uom_qty - old_qties.get(move.id, 0), 0) < 0\
+            if move.uom_id.compare(move.product_uom_qty - old_qties.get(move.id, 0), 0) < 0\
                     and move.procure_method == 'make_to_order'\
                     and all(m.state == 'done' for m in move.move_orig_ids):
                 continue
-            if move.product_uom.compare(move.product_uom_qty, 0) > 0:
+            if move.uom_id.compare(move.product_uom_qty, 0) > 0:
                 if move._should_assign_at_confirm():
                     to_assign |= move
 
@@ -330,7 +330,7 @@ class StockMove(models.Model):
                 procurement_qty = max(procurement_qty, possible_reduceable_qty)
                 values = move._prepare_procurement_values()
                 procurements.append(self.env['stock.rule'].Procurement(
-                    move.product_id, procurement_qty, move.product_uom,
+                    move.product_id, procurement_qty, move.uom_id,
                     move.location_id, move.reference, move.origin, move.company_id, values))
 
         to_assign._action_assign()
@@ -377,10 +377,10 @@ class StockMove(models.Model):
             if not bom:
                 moves_ids_to_return.add(move.id)
                 continue
-            if move.product_uom.is_zero(move.product_uom_qty):
-                factor = move.product_uom._compute_quantity(move.quantity, bom.product_uom_id) / bom.product_qty
+            if move.uom_id.is_zero(move.product_uom_qty):
+                factor = move.uom_id._compute_quantity(move.quantity, bom.uom_id) / bom.product_qty
             else:
-                factor = move.product_uom._compute_quantity(move.product_uom_qty, bom.product_uom_id) / bom.product_qty
+                factor = move.uom_id._compute_quantity(move.product_uom_qty, bom.uom_id) / bom.product_qty
             _dummy, lines = bom.sudo().explode(move.product_id, factor, picking_type=bom.picking_type_id, never_attribute_values=move.never_product_template_attribute_value_ids)
             phantom_moves_vals_list += move._generate_all_phantom_moves(lines)
             # delete the move with original product which is not relevant anymore
@@ -443,7 +443,7 @@ class StockMove(models.Model):
         return {
             'picking_id': self.picking_id.id if self.picking_id else False,
             'product_id': bom_line.product_id.id,
-            'product_uom': bom_line.product_uom_id.id,
+            'uom_id': bom_line.uom_id.id,
             'product_uom_qty': product_qty,
             'quantity': quantity_done,
             'picked': self.picked,
@@ -455,7 +455,7 @@ class StockMove(models.Model):
         self.ensure_one()
         phantom_moves_vals_list = []
         for bom_line, line_data in exploded_lines_data:
-            if self.product_uom.is_zero(self.product_uom_qty) or self.env.context.get('is_scrap'):
+            if self.uom_id.is_zero(self.product_uom_qty) or self.env.context.get('is_scrap'):
                 vals = self._generate_move_phantom(bom_line, 0, line_data['qty'])
             else:
                 vals = self._generate_move_phantom(bom_line, line_data['qty'], 0)
@@ -511,7 +511,7 @@ class StockMove(models.Model):
         if self.state in ('done', 'cancel'):
             return True
         # Do not update extra product quantities
-        return self.product_uom.is_zero(self.product_uom_qty)
+        return self.uom_id.is_zero(self.product_uom_qty)
 
     def _prepare_move_line_vals(self, quantity=None, reserved_quant=None):
         vals = super()._prepare_move_line_vals(quantity, reserved_quant)
@@ -553,7 +553,7 @@ class StockMove(models.Model):
 
         def get_qty(move):
             if move.picked:
-                return move.product_uom._compute_quantity(move.quantity, move.product_id.uom_id, rounding_method='HALF-UP')
+                return move.uom_id._compute_quantity(move.quantity, move.product_id.uom_id, rounding_method='HALF-UP')
             else:
                 return move.product_qty
 
@@ -561,7 +561,7 @@ class StockMove(models.Model):
             # skip service since we never deliver them
             if bom_line.product_id.type == 'service':
                 continue
-            if bom_line.product_uom_id.is_zero(bom_line_data['qty']):
+            if bom_line.uom_id.is_zero(bom_line_data['qty']):
                 # As BoMs allow components with 0 qty, a.k.a. optionnal components, we simply skip those
                 # to avoid a division by zero.
                 continue
@@ -571,7 +571,7 @@ class StockMove(models.Model):
                 # Then, we collect every relevant moves related to a specific component
                 # to know how many are considered delivered.
                 uom_qty_per_kit = bom_line_data['qty'] / (bom_line_data['original_qty'])
-                qty_per_kit = bom_line.product_uom_id._compute_quantity(uom_qty_per_kit / kit_bom.product_qty, bom_line.product_id.uom_id, round=False)
+                qty_per_kit = bom_line.uom_id._compute_quantity(uom_qty_per_kit / kit_bom.product_qty, bom_line.product_id.uom_id, round=False)
                 if not qty_per_kit:
                     continue
                 incoming_qty = sum(bom_line_moves.filtered(filters['incoming_moves']).mapped(get_qty))
@@ -634,8 +634,8 @@ class StockMove(models.Model):
         res = super()._get_relevant_state_among_moves()
         if res == 'partially_available'\
                 and self.raw_material_production_id\
-                and all(move.should_consume_qty and move.product_uom.compare(move.quantity, move.should_consume_qty) >= 0
-                        or (move.product_uom.compare(move.quantity, move.product_uom_qty) >= 0 or (move.manual_consumption and move.picked))
+                and all(move.should_consume_qty and move.uom_id.compare(move.quantity, move.should_consume_qty) >= 0
+                        or (move.uom_id.compare(move.quantity, move.product_uom_qty) >= 0 or (move.manual_consumption and move.picked))
                         for move in self):
             res = 'assigned'
         return res

--- a/addons/mrp/models/stock_orderpoint.py
+++ b/addons/mrp/models/stock_orderpoint.py
@@ -61,7 +61,7 @@ class StockWarehouseOrderpoint(models.Model):
             values['bom'] = self.bom_id
         return values
 
-    @api.depends('bom_id', 'bom_id.product_uom_id', 'product_id.bom_ids', 'product_id.bom_ids.product_uom_id')
+    @api.depends('bom_id', 'bom_id.uom_id', 'product_id.bom_ids', 'product_id.bom_ids.uom_id')
     def _compute_qty_to_order_computed(self):
         """ Extend to add more depends values """
         super()._compute_qty_to_order_computed()
@@ -70,7 +70,7 @@ class StockWarehouseOrderpoint(models.Model):
         super()._compute_allowed_replenishment_uom_ids()
         for orderpoint in self:
             if 'manufacture' in orderpoint.rule_ids.mapped('action'):
-                orderpoint.allowed_replenishment_uom_ids += orderpoint.product_id.bom_ids.product_uom_id
+                orderpoint.allowed_replenishment_uom_ids += orderpoint.product_id.bom_ids.uom_id
 
     def _compute_show_supply_warning(self):
         for orderpoint in self:
@@ -147,7 +147,7 @@ class StockWarehouseOrderpoint(models.Model):
         if not any(r.action == 'manufacture' for r in routes.rule_ids):
             return super()._get_replenishment_multiple_alternative(qty_to_order)
         bom = self.bom_id or self.env['mrp.bom']._bom_find(self.product_id, picking_type=False, bom_type='normal', company_id=self.company_id.id)[self.product_id]
-        return bom.product_uom_id
+        return bom.uom_id
 
     def _quantity_in_progress(self):
         bom_kits = self.env['mrp.bom']._bom_find(self.product_id, bom_type='phantom')
@@ -165,10 +165,10 @@ class StockWarehouseOrderpoint(models.Model):
             ratios_total = []
             for bom_line, bom_line_data in bom_sub_lines:
                 component = bom_line.product_id
-                if not component.is_storable or bom_line.product_uom_id.is_zero(bom_line_data['qty']):
+                if not component.is_storable or bom_line.uom_id.is_zero(bom_line_data['qty']):
                     continue
                 uom_qty_per_kit = bom_line_data['qty'] / bom_line_data['original_qty']
-                qty_per_kit = bom_line.product_uom_id._compute_quantity(uom_qty_per_kit, bom_line.product_id.uom_id, raise_if_failure=False)
+                qty_per_kit = bom_line.uom_id._compute_quantity(uom_qty_per_kit, bom_line.product_id.uom_id, raise_if_failure=False)
                 if not qty_per_kit:
                     continue
                 qty_by_product_location, dummy = component._get_quantity_in_progress(orderpoint.location_id.ids)
@@ -179,7 +179,7 @@ class StockWarehouseOrderpoint(models.Model):
             # For a kit, the quantity in progress is :
             #  (the quantity if we have received all in-progress components) - (the quantity using only available components)
             product_qty = min(ratios_total or [0]) - min(ratios_qty_available or [0])
-            res[orderpoint.id] = orderpoint.product_id.uom_id._compute_quantity(product_qty, orderpoint.product_uom, round=False)
+            res[orderpoint.id] = orderpoint.product_id.uom_id._compute_quantity(product_qty, orderpoint.uom_id, round=False)
 
         bom_manufacture = self.env['mrp.bom']._bom_find(orderpoints_without_kit.product_id, bom_type='normal')
         bom_manufacture = self.env['mrp.bom'].concat(*bom_manufacture.values())
@@ -191,11 +191,11 @@ class StockWarehouseOrderpoint(models.Model):
                 ('orderpoint_id', 'in', orderpoints_without_kit.ids),
                 ('id', 'not in', self.env.context.get('ignore_mo_ids', [])),
             ],
-            ['orderpoint_id', 'product_uom_id'],
+            ['orderpoint_id', 'uom_id'],
             ['product_qty:sum'])
         for orderpoint, uom, product_qty_sum in productions_group:
             res[orderpoint.id] += uom._compute_quantity(
-                product_qty_sum, orderpoint.product_uom, round=False)
+                product_qty_sum, orderpoint.uom_id, round=False)
 
         # add quantities coming from confirmed MO to be started but not finished
         # by the end of the stock forecast
@@ -209,8 +209,8 @@ class StockWarehouseOrderpoint(models.Model):
             date_start, date_finished, orderpoint = prod.date_start, prod.date_finished, prod.orderpoint_id
             lead_horizon_date = datetime.combine(orderpoint.lead_horizon_date, time.max)
             if date_start <= lead_horizon_date < date_finished:
-                res[orderpoint.id] += prod.product_uom_id._compute_quantity(
-                        prod.product_qty, orderpoint.product_uom, round=False)
+                res[orderpoint.id] += prod.uom_id._compute_quantity(
+                        prod.product_qty, orderpoint.uom_id, round=False)
         return res
 
     def _prepare_procurement_values(self, date=False):

--- a/addons/mrp/models/stock_rule.py
+++ b/addons/mrp/models/stock_rule.py
@@ -53,11 +53,11 @@ class StockRule(models.Model):
         for procurement in procurements:
             bom_kit = kits_by_company[procurement.company_id].get(procurement.product_id)
             if bom_kit:
-                order_qty = procurement.product_uom._compute_quantity(procurement.product_qty, bom_kit.product_uom_id, round=False)
+                order_qty = procurement.uom_id._compute_quantity(procurement.product_qty, bom_kit.uom_id, round=False)
                 qty_to_produce = (order_qty / bom_kit.product_qty)
                 _dummy, bom_sub_lines = bom_kit.explode(procurement.product_id, qty_to_produce, never_attribute_values=procurement.values.get("never_product_template_attribute_value_ids"))
                 for bom_line, bom_line_data in bom_sub_lines:
-                    bom_line_uom = bom_line.product_uom_id
+                    bom_line_uom = bom_line.uom_id
                     quant_uom = bom_line.product_id.uom_id
                     # recreate dict of values since each child has its own bom_line_id
                     values = dict(procurement.values, bom_line_id=bom_line.id)
@@ -81,7 +81,7 @@ class StockRule(models.Model):
     def _run_manufacture(self, procurements):
         new_productions_values_by_company = defaultdict(lambda: defaultdict(list))
         for procurement, rule in procurements:
-            if procurement.product_uom.compare(procurement.product_qty, 0) <= 0:
+            if procurement.uom_id.compare(procurement.product_qty, 0) <= 0:
                 # If procurement contains negative quantity, don't create a MO that would be for a negative value.
                 continue
             bom = rule._get_matching_bom(procurement.product_id, procurement.company_id, procurement.values)
@@ -93,20 +93,20 @@ class StockRule(models.Model):
             is_batch_size = bom and bom.enable_batch_size
             if not mo or is_batch_size:
                 procurement_qty = procurement.product_qty
-                batch_size = bom.product_uom_id._compute_quantity(bom.batch_size, procurement.product_uom) if is_batch_size else procurement_qty
+                batch_size = bom.uom_id._compute_quantity(bom.batch_size, procurement.uom_id) if is_batch_size else procurement_qty
                 vals = rule._prepare_mo_vals(*procurement, bom)
-                while procurement.product_uom.compare(procurement_qty, 0) > 0:
+                while procurement.uom_id.compare(procurement_qty, 0) > 0:
                     new_productions_values_by_company[procurement.company_id.id]['values'].append({
                         **vals,
-                        'product_qty': procurement.product_uom._compute_quantity(batch_size, bom.product_uom_id) if bom else procurement_qty,
+                        'product_qty': procurement.uom_id._compute_quantity(batch_size, bom.uom_id) if bom else procurement_qty,
                     })
                     new_productions_values_by_company[procurement.company_id.id]['procurements'].append(procurement)
                     procurement_qty -= batch_size
             else:
-                procurement_product_uom_qty = procurement.product_uom._compute_quantity(procurement.product_qty, procurement.product_id.uom_id)
+                procurement_product_uom_qty = procurement.uom_id._compute_quantity(procurement.product_qty, procurement.product_id.uom_id)
                 self.env['change.production.qty'].sudo().with_context(skip_activity=True).create({
                     'mo_id': mo.id,
-                    'product_qty': mo.product_id.uom_id._compute_quantity((mo.product_uom_qty + procurement_product_uom_qty), mo.product_uom_id),
+                    'product_qty': mo.product_id.uom_id._compute_quantity((mo.product_uom_qty + procurement_product_uom_qty), mo.uom_id),
                 }).change_prod_qty()
 
         for company_id in new_productions_values_by_company:
@@ -173,8 +173,8 @@ class StockRule(models.Model):
             'product_id': product_id.id,
             'product_description_variants': values.get('product_description_variants'),
             'never_product_template_attribute_value_ids': values.get('never_product_template_attribute_value_ids'),
-            'product_qty': product_uom._compute_quantity(product_qty, bom.product_uom_id) if bom else product_qty,
-            'product_uom_id': bom.product_uom_id.id if bom else product_uom.id,
+            'product_qty': product_uom._compute_quantity(product_qty, bom.uom_id) if bom else product_qty,
+            'uom_id': bom.uom_id.id if bom else product_uom.id,
             'location_src_id': picking_type.default_location_src_id.id,
             'location_dest_id': picking_type.default_location_dest_id.id or location_dest_id.id,
             'location_final_id': location_dest_id.id,
